pkg/dns/aws/dns.go

```diff
@@ -255,70 +257,85 @@ func (m *Provider) change(record *iov1.DNSRecord, zone configv1.DNSZone, action
 		return fmt.Errorf("failed to find hosted zone for record: %v", err)
 	}
 
-	// Find the target hosted zone of the load balancer attached to the service.
-	targetHostedZoneID, err := m.getLBHostedZone(target)
-	if err != nil {
-		return fmt.Errorf("failed to get hosted zone for load balancer target %q: %v", target, err)
-	}
+	var changes []*route53.Change
 
-	// Configure records and cache updates.
-	// TODO: handle the caching/diff detection in a better way.
-	m.lock.Lock()
-	defer m.lock.Unlock()
-	key := zoneID + domain + target
-	// Only process updates once for now because we're not diffing.
-	if m.updatedRecords.Has(key) && action == upsertAction {
-		log.V(2).Info("skipping cached DNS record update", "record", record)
-		return nil
-	}
-	err = m.updateAlias(domain, zoneID, target, targetHostedZoneID, string(action))
-	if err != nil {
-		return fmt.Errorf("failed to update alias in zone %s: %v", zoneID, err)
+	for _, target := range targets {
+		// Find the target hosted zone of the load balancer attached to the service.
+		targetHostedZoneID, err := m.getLBHostedZone(target)
+		if err != nil {
+			return fmt.Errorf("failed to get hosted zone for load balancer target %q: %v", target, err)
+		}
+		key := zoneID + domain + target
+		// Configure records and cache updates.
+		// TODO: handle the caching/diff detection in a better way.
+		m.lock.Lock()
+		alreadyHaveRecord := updatedRecords.Has(key)
+		m.lock.Unlock()
+		// Only process updates once for now because we're not diffing.
+		if alreadyHaveRecord && action == upsertAction {
+			log.V(2).Info("skipping cached DNS record update", "record", record)
+			continue
+		}
+
+		switch action {
+		case upsertAction:
+			updatedRecords.Insert(key)
+			log.Info("upserting DNS record", "record", record)
+		case deleteAction:
+			updatedRecords.Delete(key)
+			log.Info("deleting DNS record", "record", record)
+		}
+
+		change := &route53.Change{
+			Action: aws.String(string(action)),
+			ResourceRecordSet: &route53.ResourceRecordSet{
+				Name: aws.String(domain),
+				Type: aws.String("A"),
+				AliasTarget: &route53.AliasTarget{
+					HostedZoneId:         aws.String(targetHostedZoneID),
+					DNSName:              aws.String(target),
+					EvaluateTargetHealth: aws.Bool(false),
+				},
+			},
+		}
+
+		changes = append(changes, change)
 	}
-	switch action {
-	case upsertAction:
-		m.updatedRecords.Insert(key)
-		log.Info("upserted DNS record", "record", record)
-	case deleteAction:
-		m.updatedRecords.Delete(key)
-		log.Info("deleted DNS record", "record", record)
+
+	if len(changes) > 0 {
+		err = m.updateAlias(domain, zoneID, targets, string(action), changes)
+		if err != nil {
+			return fmt.Errorf("failed to update alias in zone %s: %v", zoneID, err)
+		} else {
```

Comment:
You don't need `else` since the `if` block ends with a `return`.

Reply:
done